firedrake/utils.py

```diff
@@ -74,3 +77,20 @@ def wrapper(f, *args, **kwargs):
         finally:
             opts["type_check"] = check
     return decorator(wrapper, f)
+
+@functools.lru_cache(maxsize=None)
+def get_eigen_include_dir():
+    """Return the include directory for Eigen.
+    
+    Depending on how Eigen was installed this will either be defined in
+    petscvariables or the Firedrake configuration file.
+
+    Returns ``None`` if not found.
+    """
+    try:
+        return get_petsc_variables()["EIGEN_INCLUDE_DIR"].lstrip("-I")
```

Comment:
So we call `get_petsc_variables` again here. Can we leave the call in the old line 62 and just pass the result of `get_petsc_variables` in this function as a parameter, i.e. sth like `get_eigen_include_dir(petsc_variables)`

Reply:
That could be done but I don't really like structuring the code like that. Getters aren't supposed to take arguments.

Reply:
I think the reason for why getters are supposed to not take arguments in general is because they are usually attached to a class or an object that is already having the information stored on an attribute. That's not the case here. But in any case this is fine, I don't have a strong opinion about it.